GoogleDataTransport/GDTCCTLibrary/GDTCCTUploadOperation.m

```diff
@@ -518,13 +557,14 @@ - (void)main {
 }
 
 - (void)cancel {
-  GDTCORLogDebug(@"Upload operation cancelled: %@", self);
-  [super cancel];
+  @synchronized(self) {
+    [super cancel];
 
-  // If the operation hasn't been started we can set `isFinished = YES` straight away.
-  if (!_executing) {
-    _executing = NO;
-    _finished = YES;
+    // If the operation hasn't been started we can set `isFinished = YES` straight away.
+    if (!_executing) {
+      _executing = NO;
+      _finished = YES;
```

Comment:
Question, should this be these two lines be prefixed with `self->`? I see they are on on lines 545 & 546.

Reply:
Good question. Usually explicit self is required when ivars (instance variables) are used in a block, so this type of usage will implicitly lead to capturing `self` by the block. Compiler warns to add explicit `self->` to "unhide" this capturing and make it more visible to the code reader. Here the ivars are not captured by a block, so accessing them directly is fine.

@paulb777 @ryanwilson Do you have anything to add/correct here?

Reply:
SGTM